enhancements/installer/image-based-installer.md

```diff
@@ -0,0 +1,266 @@
+the cluster to be installed provided as input.
+The Config Image Contents:
+* ClusterInfo (cluster name, base domain, hostname, nodeIP)
+* SSH authorized_keys
+* Pull Secret
+* Extra Manifests
+* Generated keys and certs (compatible the generated admin kubeconfig)
+* Static networking config
+
+The site specific configuration data will be generated according to information provided
+in the install-config.yaml and the manifests provided in the installation directory as input.
+To complete the installation at the edge site:
+The cluster configuration for the edge location can be delivered by copying the config iso content onto the node
+and placing it under /opt/openshift/cluster-configuration/
+The cluster configuration can also b delivered using an attached ISO, a systemd service running on the host
+pre-installed Image-based Installer will mount that ISO (identified by known label) and copy the cluster configuration
+to /opt/openshift/cluster-configuration/
+The cluster configuration data on the disk will be used to configure the cluster and allow OCP to start successfully.
+
+### Workflow Description
+
+TBD
+
+### API Extensions
+
+N/A
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+The image-based installer targets single-node OpenShift deployments.
+
+### Implementation Details/Notes/Constraints
+
+Since we must allow users to provision hosts themselves, either manually or
+using automated tooling of their choice, the ISO format offers the widest range
+of compatibility. Building a single ISO to boot multiple hosts makes it
+considerably easier for the user to manage. The additional site configuration
+ISO is necessary for configuring each cluster securely and independently.
+
+The user, before running the image-based installer, must generate a [seed image](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md)
+via the [Lifecycle Agent SeedGenerator Custom Resouce (CR)](https://github.com/openshift-kni/lifecycle-agent/blob/main/docs/seed-image-generation.md).
+The prerequisites to generating a seed image are the following:
+
+- An already provisioned single-node OpenShift cluster (seed SNO).
+   - The CPU topology of that host must align with the target host(s), i.e. they
+     should have the same number of cores.
+- The [Lifecycle Agent](https://github.com/openshift-kni/lifecycle-agent/tree/main)
+  operator must be installed on the seed SNO.
+
+### Risks and Mitigations
+
+N/A
+
+### Drawbacks
+
+N/A
+
+## Open Questions [optional]
+
+- Should the command-line tool that generates the installer ISO be a subcommand of the OpenShift installer, or a
+  standalone binary?
+
+  Having the functionality provided by the command-line tool in the OpenShift
+  installer would be a natural addition to the latter, as the former refers to
```

Comment:
It's worth noting that a key point of the installer is that it shares a release cycle with OCP.

So if you want the ISO building tool to be locked to an OCP version (i.e. you use the 4.17 installer to build an ISO to install a 4.17 seed) then you'll want to be in the installer.

If you want to always use the latest image builder to install regardless of which release the seed came from, you will come to regret putting it in the installer.

Reply:
> It's worth noting that a key point of the installer is that it shares a release cycle with OCP.

That's a good point, we currently don't have strict requirements for either the installer `locked` OCP version or the `latest` one (regardless of the seed OCI image OCP version). But in the future it's probable that we lean towards the former.

 > Everything else in the installer takes an install-config as its input and operates on a shared asset graph stored in the same asset store. In this case, even the 2 ibi commands do not operate on the same assets.
 
Both the IBI configuration ISO and installation ISO generation commands take an `install-config.yaml` (and additional image-based configs) as their input. But you are correct that they don't operate on shared assets.

(please do not take into consideration the installation ISO generation input you see [here](https://github.com/openshift/installer/pull/8422), we need parts of the `install-config.yaml` as well. That PR is going to be reworked to reflect the actual flow.)

Reply:
> > Both the IBI configuration ISO and installation ISO generation commands take an install-config.yaml

> Wait, what? This is news to me.
What does the installation ISO use the install-config for? I thought it would pull all the data it needed from the seed image.

We don't have to use the `install-config.yaml` and instead include everything we need in the image-based configuration of the installation ISO generation command. This seems to resolve all the concerns you mentioned IIUC, right?